src/databricks/labs/remorph/transpiler/lsp/lsp_engine.py

```diff
@@ -205,11 +205,12 @@ def register_capabilities(self, params: RegistrationParams) -> None:
 
     async def transpile_document_async(self, params: TranspileDocumentParams) -> TranspileDocumentResult:
         if self.stopped:
-            raise RuntimeError("Client has been stopped.")
-        await self._await_for_transpile_capability()
+            raise IllegalStateException("Client has been stopped.")
+        if not self.transpile_to_databricks_capability:
+            raise IllegalStateException("Client has not yet registered its transpile capability.")
         return await self.protocol.send_request_async(TRANSPILE_TO_DATABRICKS_METHOD, params)
 
-    async def _await_for_transpile_capability(self):
+    async def await_for_transpile_capability(self):
```

Comment:
This method shouldn't be exposed

Reply:
I've refactored this section; it's no longer exposed.